### PR TITLE
Flightapp

### PR DESCRIPTION
**Pull request by visha15l**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -33,6 +39,9 @@ button.booking {
   box-shadow: 2px 2px 2px #9e9e9e;
   margin: 5px;
 }
+.filter {
```
Comment (visha15l) [inline src/App.css:40]: class name should be .center

File: src/App.css [hunk 2]
```diff
@@ -2,6 +2,12 @@
   margin: 5px 0;
   width: 100%;
 }
+.left {
+  float: left;
+}
+.right {
+  float: right;
+}
```
Comment (visha15l) [inline src/App.css:8]: do not use float use flexbox

File: src/App.css [hunk 3]
```diff
@@ -2,6 +2,12 @@
   margin: 5px 0;
   width: 100%;
 }
+.left {
+  float: left;
+}
```
Comment (visha15l) [inline src/App.css]: do not use float. use flex

File: src/App.css [hunk 4]
```diff
@@ -59,6 +68,10 @@ button.favourite {
 .result-header {
   border-bottom: 2px solid #dee2e6;
   padding: 5px 10px;
+  content: "";
+  clear: both;
```
Comment (visha15l) [inline src/App.css:84]: use flex

File: src/App.js [hunk 5]
```diff
@@ -121,15 +121,17 @@ const App = () => {
                 </Tab>
               </Tabs>
             </div>
+            <h4 className="filter">Price Filter</h4>
             <Rangeslider value={pricechange} />
-            <Link to="/favourite">Favourite</Link>
+            {/* <Link to="/favourite">Favourite</Link> */}
```
Comment (visha15l) [inline src/App.js]: why it is commented

File: src/common/favaccordian.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+const Favaccordion = ({ children, props, sdata }) => {
+  const [isOpen, setOpen] = useState(false);
+  let d1 = moment(props.data.departureTime);
+  let d2 = moment(sdata.arrivalTime);
+  var startTime = moment(props.data.departureTime, "hh:mm:ss");
+  var endTime = moment(sdata.arrivalTime, "hh:mm:ss");
+  var diff = endTime.diff(startTime, "hours");
+  var totalMinutes = endTime.diff(startTime, "minutes");
+  var clearMinutes = totalMinutes % 60;
+  const addtofavourite = () => {
```
Comment (visha15l) [inline src/common/favaccordian.js:12]: do not use var

File: src/common/favaccordian.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
+import { useState } from "react";
+import * as moment from "moment";
+const Favaccordion = ({ children, props, sdata }) => {
+  const [isOpen, setOpen] = useState(false);
+  let d1 = moment(props.data.departureTime);
+  let d2 = moment(sdata.arrivalTime);
```
Comment (visha15l) [inline src/common/favaccordian.js]: use descriptive variable names d1, d2 is not acceptable

File: src/common/favaccordian.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  var startTime = moment(props.data.departureTime, "hh:mm:ss");
+  var endTime = moment(sdata.arrivalTime, "hh:mm:ss");
+  var diff = endTime.diff(startTime, "hours");
+  var totalMinutes = endTime.diff(startTime, "minutes");
+  var clearMinutes = totalMinutes % 60;
+  const addtofavourite = () => {
+    var newArray = JSON.parse(sessionStorage.getItem("fav"));
+    //var newArray = [sessionStorage.getItem("fav")];
+    newArray = newArray ? newArray : [];
+    //console.log(newArray);
```
Comment (visha15l) [inline src/common/favaccordian.js]: remove console.log

File: src/common/favaccordian.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    newArray = [...newArray, props];
+    sessionStorage.setItem("fav", JSON.stringify(newArray));
+    console.log(JSON.parse(sessionStorage.getItem("fav")));
```
Comment (visha15l) [inline src/common/favaccordian.js]: remove console.log

File: src/common/favaccordian.js [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+  };
+  const clear = () => {
+    sessionStorage.clear();
+  };
+  if (diff > 0) {
```
Comment (visha15l) [inline src/common/favaccordian.js]: use conditional operator to render content . we do not use if else for rendering.

File: src/common/favaccordian.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    return (
+      <div className="accordion-wrapper">
+        <div className={`accordion-title ${isOpen ? "open" : ""}`}>
```
Comment (visha15l) [inline src/common/favaccordian.js]: use clsx npm package to handle classnames

File: src/common/favaccordian.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
+import { useState } from "react";
+import * as moment from "moment";
+const Favaccordion = ({ children, props, sdata }) => {
```
Comment (visha15l) [inline src/common/favaccordian.js]: what is sdata?

File: src/common/favmulti.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
+import logo from "../images.png";
+import Favaccordion from "../common/favaccordian";
+import Multisingle from "../common/multisingle";
+const Favmulti = (props) => {
+  return (
+    <Favaccordion props={props} sdata={props.sdata}>
+      <Multisingle flight={props.data} />
+      <Multisingle flight={props.sdata} />
```
Comment (visha15l) [inline src/common/favmulti.js:8]: why are you using same component twice?

File: src/component/favourite/favourite.js [hunk 14]
```diff
@@ -0,0 +1,44 @@
+import _ from "lodash";
+import { Component } from "react";
+import Favdirectrow from "../../common/favdirectrow";
+import Favmulti from "../../common/favmulti";
+import { Link } from "react-router-dom";
+class Favourite extends Component {
+  render() {
+    const renderfavflights = () => {
+      let data = JSON.parse(sessionStorage.getItem("dfav"));
+      console.log([JSON.parse(sessionStorage.getItem("dfav"))]);
```
Comment (visha15l) [inline src/component/favourite/favourite.js]: remove console.log

File: src/component/favourite/favourite.js [hunk 15]
```diff
@@ -0,0 +1,44 @@
+import _ from "lodash";
+import { Component } from "react";
+import Favdirectrow from "../../common/favdirectrow";
+import Favmulti from "../../common/favmulti";
+import { Link } from "react-router-dom";
+class Favourite extends Component {
```
Comment (visha15l) [inline src/component/favourite/favourite.js:6]: use react hooks

File: src/component/favourite/favourite.js [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+          <Favdirectrow
+            key={plane.data.flightNo + plane.data.origin}
+            data={plane.data}
+          />
+        );
+      });
+    };
+    const renderfavmultiflights = () => {
+      let multi = JSON.parse(window.sessionStorage.getItem("fav"));
+      console.log(multi);
```
Comment (visha15l) [inline src/component/favourite/favourite.js]: remove console.log

File: src/component/search/tabs/tabbutton.js [hunk 17]
```diff
@@ -0,0 +1,11 @@
...
+   
+    return(
+      <div className="tab-buttons">
+      {buttons.map(button =>{
+         return <button className={button === activeTab? 'tab active': 'tab inactive'} onClick={()=>changeTab(button)}>{button}</button>
+      })}
+      </div>
+    )
+  }
+  export default TabButtons;
```
Comment (visha15l) [inline src/component/search/tabs/tabbutton.js:11]: please format the file.

File: src/component/search/tabs/tabs.js [hunk 18]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import TabButtons from './tabbutton';
+class Tabs extends React.Component{
```
Comment (visha15l) [inline src/component/search/tabs/tabs.js:3]: use react hooks

File: src/component/search/tabs/tabs.js [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+          })}
+           
+          <TabButtons activeTab={this.state.activeTab} buttons={buttons} changeTab={this.changeTab}/>
+          <div className="tab-content">{content}</div>
+          
+        </div>
+      );
+    }
+  }
+  export default Tabs;
```
Comment (visha15l) [inline src/component/search/tabs/tabs.js:29]: please format this file.

File: src/component/searchresult/accordian.js [hunk 20]
```diff
@@ -10,10 +10,14 @@ const Accordion = ({ children, props, sdata }) => {
...
   var clearMinutes = totalMinutes % 60;
-  const favourite = () => {
-    var newArray = [JSON.parse(window.sessionStorage.getItem("fav"))];
-    newArray.push(props);
+  const addtofavourite = () => {
+    var newArray = JSON.parse(sessionStorage.getItem("fav"));
+    //var newArray = [sessionStorage.getItem("fav")];
+    newArray = newArray ? newArray : [];
+    //console.log(newArray);
+    newArray = [...newArray, props];
```
Comment (visha15l) [inline src/component/searchresult/accordian.js:14]: remove console.logs

File: src/component/searchresult/index.js [hunk 21]
```diff
@@ -30,10 +30,11 @@ const Searchresult = (props) => {
...
             props.data.Destination.toLowerCase() ||
-          (v.origin.toLowerCase() === props.data.Origin.toLowerCase() &&
-            v.date === props.data.D_Date &&
-            v.price <= props.filtervalues.maxValue &&
-            v.price >= props.filtervalues.minValue)
+          v.origin.toLowerCase() === props.data.Origin.toLowerCase()
+        // &&
+        //   v.date === props.data.D_Date &&
+        //   v.price <= props.filtervalues.maxValue &&
+        //   v.price >= props.filtervalues.minValue
```
Comment (visha15l) [inline src/component/searchresult/index.js]: remove console.log

File: src/component/searchresult/index.js [hunk 22]
```diff
@@ -30,10 +30,11 @@ const Searchresult = (props) => {
         (v) =>
```
Comment (visha15l) [inline src/component/searchresult/index.js:30]: what is v ?

File: src/component/searchresult/index.js [hunk 23]
```diff
@@ -71,9 +73,10 @@ const Searchresult = (props) => {
...
           v.destination.toLowerCase() === props.data.Origin.toLowerCase() ||
           (v.origin.toLowerCase() === props.data.Destination.toLowerCase() &&
-            v.date === props.data.R_Date &&
-            v.price <= props.filtervalues.maxValue &&
-            v.price >= props.filtervalues.minValue)
+            v.date === props.data.R_Date)
+        // &&
+        // v.price <= props.filtervalues.maxValue &&
+        // v.price >= props.filtervalues.minValue
```
Comment (visha15l) [inline src/component/searchresult/index.js]: remove console.log

File: src/component/searchresult/index.js [hunk 24]
```diff
@@ -98,6 +101,7 @@ const Searchresult = (props) => {
       ); //second of multiflight return
       setsecondreturnData(secondreturn);
     };
+    console.log(props);
```
Comment (visha15l) [inline src/component/searchresult/index.js]: remove console.log

File: src/component/searchresult/index.js [hunk 25]
```diff
@@ -119,47 +123,80 @@ const Searchresult = (props) => {
   };
   const rendermultiflights = () => {
     let secondf;
```
Comment (visha15l) [inline src/component/searchresult/index.js:114]: what is secondf?

File: src/component/searchresult/index.js [hunk 26]
```diff
@@ -119,47 +123,80 @@ const Searchresult = (props) => {
...
         return obj.origin === plane.destination;
       });
-      return (
-        <Multirow
-          key={plane.flightNo + plane.origin}
-          data={plane}
-          sdata={secondf}
-        />
-      );
+      console.log(secondf);
```
Comment (visha15l) [inline src/component/searchresult/index.js]: remove console.log

File: src/constants/state.js [hunk 27]
```diff
@@ -0,0 +1,7 @@
+export const ONEWAY_STATE = {
+  Origin: '', Destination: '', Date: '', count:''
+  };
+  export const TWOWAY_STATE = {
+    Origin: '', Destination: '', DepartureDate: '',ReturnDate:'', count:''
+    };
+
```
Comment (visha15l) [inline src/constants/state.js]: please format this file.

File: src/common/favmulti.js (hunk 13 above)
Comment (Sonalicodeastu) [reply src/common/favmulti.js:8]: for the multiflight  first component for first flight and second for second flight

File: src/component/search/tabs/tabs.js (hunk 19 above)
Comment (Sonalicodeastu) [reply src/component/search/tabs/tabs.js:29]: file is  already getting formatted automatically

File: src/component/searchresult/index.js (hunk 25 above)
Comment (Sonalicodeastu) [reply src/component/searchresult/index.js:114]: second flight of multiflight

File: src/common/favaccordian.js [hunk 28]
```diff
@@ -1,71 +1,71 @@
...
-    return (
-      <div className="accordion-wrapper">
-        <div className={`accordion-title ${isOpen ? "open" : ""}`}>
-          <div className="row flex-container">
-            <div className="borderbox">
-              <div>
-                <span>Multiple</span>
+  return (
+    <div>
+      {diff ? (
```
Comment (visha15l) [inline src/common/favaccordian.js:23]: use diff > 0

File: src/common/favaccordian.js [hunk 29]
```diff
@@ -1,71 +1,71 @@
...
+        <div className="accordion-wrapper">
+          <div className={`accordion-title ${isOpen ? "open" : ""}`}>
+            <div className="row flex-container">
+              <div className="borderbox">
+                <div>
```
Comment (visha15l) [inline src/common/favaccordian.js:28]: why you are using empty div here? you can make span element as a block level element using css.

File: src/common/favaccordian.js [hunk 30]
```diff
@@ -1,71 +1,71 @@
...
+                  <span>Multiple</span>
+                </div>
+                <div>
+                  <a href="#" onClick={() => setOpen(!isOpen)}>
```
Comment (visha15l) [inline src/common/favaccordian.js:32]: we do not use href="#" in react.

File: src/component/searchresult/accordian.js [hunk 31]
```diff
@@ -5,71 +5,71 @@ const Accordion = ({ children, props, sdata }) => {
...
-    return (
-      <div className="accordion-wrapper">
-        <div className={`accordion-title ${isOpen ? "open" : ""}`}>
-          <div className="row flex-container">
-            <div className="borderbox">
-              <div>
-                <span>Multiple</span>
+  return (
+    <div>
+      {diff ? (
```
Comment (visha15l) [inline src/component/searchresult/accordian.js:22]: use diff > 0

File: src/component/searchresult/accordian.js [hunk 32]
```diff
@@ -5,71 +5,71 @@ const Accordion = ({ children, props, sdata }) => {
...
+        <div className="accordion-wrapper">
+          <div className={`accordion-title ${isOpen ? "open" : ""}`}>
+            <div className="row flex-container">
+              <div className="borderbox">
+                <div>
```
Comment (visha15l) [inline src/component/searchresult/accordian.js:27]: remove this div

File: src/component/searchresult/accordian.js [hunk 33]
```diff
@@ -5,71 +5,71 @@ const Accordion = ({ children, props, sdata }) => {
...
+                  <span>Multiple</span>
+                </div>
+                <div>
+                  <a href="#" onClick={() => setOpen(!isOpen)}>
```
Comment (visha15l) [inline src/component/searchresult/accordian.js:31]: remove "#"

File: src/component/searchresult/index.js [hunk 34]
```diff
@@ -31,10 +31,6 @@ const Searchresult = (props) => {
           v.destination.toLowerCase() ===
             props.data.Destination.toLowerCase() ||
           v.origin.toLowerCase() === props.data.Origin.toLowerCase()
```
Comment (visha15l) [inline src/component/searchresult/index.js]: do not use v . use flight

File: src/App.js [hunk 35]
```diff
@@ -11,23 +11,24 @@ import Rangeslider from "./component/filter/rangeslider";
...
 
 const App = () => {
   const [inputValues, setInputValues] = useState({
-    Origin: "",
-    Destination: "",
-    D_Date: "",
-    R_Date: "",
+    origin: "",
+    destination: "",
+    d_Date: "",
```
Comment (visha15l) [inline src/App.js]: please use descriptive variable names.

File: src/App.js [hunk 36]
```diff
@@ -11,23 +11,24 @@ import Rangeslider from "./component/filter/rangeslider";
...
     count: "",
   });
   const [Values, setValues] = useState({
     maxValue: 5000,
     minValue: 0,
   });
-  const [formValues, setformValues] = useState({
-    submit: false,
-  });
+  const [activetab, setactivcetab] = useState({ oneway: "true" });
```
Comment (visha15l) [inline src/App.js]: it should be activeTab and setActiveTab .

File: src/App.js [hunk 37]
```diff
@@ -11,23 +11,24 @@ import Rangeslider from "./component/filter/rangeslider";
...
+  const [formValues, setformValues] = useState({ hits: [] });
+  const [formsubmit, setFormsubmit] = useState(false);
```
Comment (visha15l) [inline src/App.js]: it should be formSubmit and setFormSubmit

File: src/App.js [hunk 38]
```diff
@@ -11,23 +11,24 @@ import Rangeslider from "./component/filter/rangeslider";
...
   });
   const [Values, setValues] = useState({
     maxValue: 5000,
     minValue: 0,
   });
-  const [formValues, setformValues] = useState({
-    submit: false,
-  });
+  const [activetab, setactivcetab] = useState({ oneway: "true" });
+  const [formValues, setformValues] = useState({ hits: [] });
```
Comment (visha15l) [inline src/App.js]: what is hits ?

File: src/App.js [hunk 39]
```diff
@@ -11,23 +11,24 @@ import Rangeslider from "./component/filter/rangeslider";
...
+  const [formsubmit, setFormsubmit] = useState(false);
+  useEffect(() => {}, [activetab, formValues]);
```
Comment (visha15l) [inline src/App.js]: why are you using empty useEffect ?

File: src/App.js [hunk 40]
```diff
@@ -37,8 +38,27 @@ const App = () => {
     const dateChanged = value.split("-").join("/");
     setInputValues({ ...inputValues, [name]: dateChanged });
   };
-  const handleSubmit = () => {
-    setformValues({ submit: true, ...inputValues });
+  const activetabhandler = (btn) => {
+    const name = btn;
+    setactivcetab({ [name]: "true" });
```
Comment (visha15l) [inline src/App.js]: please check spelling . it should be setActiveTab

File: src/App.js [hunk 41]
```diff
@@ -37,8 +38,27 @@ const App = () => {
...
+      ? !_.isEmpty(origin) &&
+          !_.isEmpty(destination) &&
+          !_.isEmpty(d_Date) &&
+          !_.isEmpty(count)
+      : !_.isEmpty(origin) &&
+          !_.isEmpty(destination) &&
+          !_.isEmpty(d_Date) &&
+          !_.isEmpty(r_Date) &&
+          !_.isEmpty(count);
+  };
```
Comment (visha15l) [inline src/App.js:67]: please optimize it

File: src/App.js [hunk 42]
```diff
@@ -56,77 +77,72 @@ const App = () => {
...
-                    />
-                    <br />
-                    <Inputpassenger placeholder="Number of passenger" />
-                    <br />
-                  </form>
-                </Tab>
-              </Tabs>
+              <button
+                className={activetab.oneway ? "active tab" : "tab false"}
+                onClick={() => activetabhandler("oneway")}
```
Comment (visha15l) [inline src/App.js]: it should be activeTabHandler

File: src/App.js [hunk 43]
```diff
@@ -56,77 +77,72 @@ const App = () => {
...
+              >
+                Oneway
+              </button>
+              <button
+                className={activetab.ret ? "active tab" : "tab false"}
+                onClick={() => activetabhandler("ret")}
```
Comment (visha15l) [inline src/App.js]: it should be activeTabHandler.

File: src/App.js [hunk 44]
```diff
@@ -56,77 +77,72 @@ const App = () => {
...
+              >
```
Comment (visha15l) [inline src/App.js:103]: please pass entire word i.e return

File: src/App.js [hunk 45]
```diff
@@ -56,77 +77,72 @@ const App = () => {
...
+                  <InputDate
+                    name="r_Date"
+                    placeholder="Return Date"
+                    handleChange={handleChange}
+                  />
+                ) : (
+                  ""
+                )}
+
+                <br />
```
Comment (visha15l) [inline src/App.js:128]: why do we need br ?

File: src/App.js [hunk 46]
```diff
@@ -56,77 +77,72 @@ const App = () => {
...
             <Rangeslider value={pricechange} />
           </aside>
           <div className="content">
             <div>
-              <Searchresult data={inputValues} filtervalues={Values} />
+              {formsubmit ? (
+                <Searchresult data={formValues} filtervalues={Values} />
+              ) : (
+                ""
+              )}
```
Comment (visha15l) [inline src/App.js:149]: we can eliminate empty else condition using && operator.

File: src/component/searchresult/accordian.js [hunk 47]
```diff
@@ -3,8 +3,9 @@ import logo from "../../images.png";
 import * as moment from "moment";
 const Accordion = ({ children, props, sdata }) => {
   const [isOpen, setOpen] = useState(false);
-  let d1 = moment(props.data.departureTime);
-  let d2 = moment(sdata.arrivalTime);
+  // let departuretime = moment(props.data.departureTime);
+  // console.log(props);
+  // let arrivalTime = moment(props.sdata.arrivalTime);
```
Comment (visha15l) [inline src/component/searchresult/accordian.js]: please remove commented code

File: src/component/searchresult/index.js [hunk 48]
```diff
@@ -26,77 +26,78 @@ const Searchresult = (props) => {
       let directreturn = {};
       let firstreturn = {};
       let secondreturn = {};
+      console.log(props);
```
Comment (visha15l) [inline src/component/searchresult/index.js]: remove console.log

File: src/component/searchresult/index.js [hunk 49]
```diff
@@ -26,77 +26,78 @@ const Searchresult = (props) => {
...
-          v.origin.toLowerCase() === props.data.Origin.toLowerCase() &&
-          v.date === props.data.D_Date &&
+            props.data.destination.toLowerCase() &&
+          v.origin.toLowerCase() === props.data.origin.toLowerCase() &&
+          v.date === props.data.d_Date &&
           v.price <= props.filtervalues.maxValue &&
           v.price >= props.filtervalues.minValue
       ); //direct flights
       multires = multiresult.filter(
         (v) =>
```
Comment (visha15l) [inline src/component/searchresult/index.js:45]: what is v ?

**Review by visha15l — COMMENTED** [review]
Do not commit commented code. 
Do not use short variable names.
Use descriptive function and variable names.

**Pull request merged**